### DISCUSSION TITLE
rabbitmq shovel socket_closed_unexpectedly

### DISCUSSION THREAD

**Discussion by rfakit** [rabbitmq/rabbitmq-server#10001] [Other] [answered]
### Describe the bug

we deploy openstack environment using openstack-ansible. 
after upgrade to RabbitMQ 3.9.28 on Erlang 25.0.4. we constantly have this kind of socket_closed_unexpectedly error in log files of rabbit.  we use shovel plugin to send events from the ha mode rabbit to stands alone rabbit cluster.
how can we debug that.

[rabbit-log.txt](https://github.com/rabbitmq/rabbitmq-server/files/13488019/rabbit-log.txt)

### Reproduction steps

we use https://github.com/openstack/openstack-ansible-rabbitmq_server version stable/xena

### Expected behavior

formally we do not have this error and shovel works fine

**Comment by michaelklishin** [comment, marked as answer]
RabbitMQ 3.9 has been [completely out of support](https://rabbitmq.com/versions.html) for several months now.

This error [has a dedicated doc section](https://rabbitmq.com/logging.html#logged-events). If you use any proxies or load balancers, they may be [closing "inactive" connections](https://rabbitmq.com/heartbeats.html#tcp-proxies). Neither Shovel nor the rest of RabbitMQ can do anything about that.

Shovels losing TCP connection is not something we have seen reported for 3.9 or later versions.
